components/v1/Buttons/ConnectWalletDesktop.tsx

```diff
@@ -56,7 +57,15 @@ const ButtonConnectWalletDesktop: FunctionComponent<ButtonConnectWalletDesktopPr
             toast.remove();
             toast.custom((t) => <ToastError>{result.error.message}</ToastError>);
             setIsConnecting(false);
-            setConnectorName(undefined);
+            return;
+        }
+
+        // Handle WalletConnect connected to wrong network, but not unsupported (e.g. connect to BSC when in Arbtirum Markets)
+        if (c instanceof WalletConnectConnector && result.data.chain.id !== selectedChain.id) {
+            disconnect();
+            toast.remove();
```

Comment:
why we need toast.remove here ? 
and why we should disconnect first ?

Reply:
1. I think there was an issue with the toast before, so we need to `toast.remove` first to make it work properly
2. So we have a case where a user is in `arbitrum markets` page and haven't connected their wallet yet. After that, they want to connect their wallet with WalletConnect, but they chose a wrong network in their wallet. In this case, the wrong network is BSC, which is actually one of our supported network. With our previous app behavior, our app will let the user's wallet to connect to BSC while in `arbitrum markets` because it is detected as a supported network. We don't want that case to happen right? So if we detect that the network is wrong but supported, we will disconnect it.